echopype/clean/impulse_noise.py

```diff
@@ -0,0 +1,463 @@
+"""
+Algorithms for masking Impulse noise.
+
+    Copyright (c) 2020 Echopy
+
+    This code is licensed under the MIT License.
+    See https://opensource.org/license/mit/ for details.
+    Original code sourced from:
+    https://github.com/open-ocean-sounding/echopy/blob/master/echopy/processing/mask_impulse.py
+
+"""
+__authors__ = [
+    "Alejandro Ariza",  # wrote ryan(), ryan_iterable(), and wang()
+    "Raluca Simedroni",  # adapted the impulse noise masking algorithms to echopype
+]
```

Comment:
Since most of my comments would be effectively rewriting the code (hence new implementations), we can modify this block once the code is in good shape.